components/org.wso2.micro.integrator.ndatasource.core/src/main/java/org/wso2/micro/integrator/ndatasource/core/DataSourceManager.java

```diff
@@ -175,11 +176,11 @@ public void initSystemDataSources() throws DataSourceException {
 
 	private void initSystemDataSource(File sysDSFile) throws DataSourceException {
 		try {
-		    JAXBContext ctx = JAXBContext.newInstance(SystemDataSourcesConfiguration.class);
-            Document doc = DataSourceUtils.convertToDocument(sysDSFile);
-            DataSourceUtils.secureResolveDocument(doc, true);
-		    SystemDataSourcesConfiguration sysDS = (SystemDataSourcesConfiguration) ctx.createUnmarshaller().
```

Comment:
DataSourceUtils.secureResolveDocument method is used in DataSourceCappDeployer class. Maybe that also needs to be replaced with DataSourceUtils.secureResolveOMElement method if this method is not functioning properly (But it would be a separate issue). WDYT?

Reply:
Yes Deploying a Datasource with Capp will not work if the password is secured but I couldn't find a doc regarding securing passwords when deploying a Datasource through capps if we are doing that we may need to add tooling support and docs as well that's why I have left it untouched.